cmd/flow-aggregator/options.go

```diff
@@ -74,11 +74,11 @@ func (o *Options) validate(args []string) error {
 	if len(args) != 0 {
 		return errors.New("no positional arguments are supported")
 	}
-	if o.config.ExternalFlowCollectorProtocol == "" {
+	if o.config.ExternalFlowCollectorAddr == "" {
 		return fmt.Errorf("IPFIX flow collector address should be provided")
 	} else {
 		// Check if it is TCP or UDP
```

Comment:
This check is same as the one that is done in cmd/antrea-agent/options.go and there is related testing. This check is duplicated in flow aggregator and testing is duplicated. Could we separate this flow collector address config check and keep it at a commonplace? I was thinking pkg/agent/flowexporter/utils.go is appropriate location. If you can think of a better alternate option, we can go with that. 

I think we can separate out the flow interval check too. Let me know what you think.

Reply:
Thanks, it's a great idea to reuse these config validation code among flow exporter and flow aggregator. Will make the change in this way.

Reply:
Hi Srikar, I we separate the flow collector address and flow interval config check code and keep them at /pkg/agent/flowexporter/utils.go. Also added unit tests for these functions in /pkg/agent/flowexporter/utils_test.go. Do you think we could delete some test cases for flow exporter and aggregator validation functions since some of the testcases have been covered separately by the unit tests in utils_test.go?

Reply:
Yes, I think the test cases flow exporter and aggregator validation functions in the cmd/* can be deleted.
All the code in flow exporter validation seems to be covered in utils_test.go. However, in the aggregator validation function [L111-L115](https://github.com/vmware-tanzu/antrea/pull/1718/files#diff-881e8cbabcf6cfcfec09e0ad632f103993d06a26cdf31d22d30cdbed14b14597R111-R115) is not covered. Maybe this can be separated as parseTrasnportProtocol like flow collector address and interval parse functions.

Reply:
Thanks, for the aggregator validation function, I found https://github.com/vmware-tanzu/antrea/blob/45c7fffba7758a8cddf61b6b64fe1596322a06ff/cmd/flow-aggregator/options.go#L87-L98 is not covered either, because we were passing the net.Addr instead of a string to flow aggregator, shall I keep the options_test.go in cmd/flow-aggregator or add a new util function named like ParseFlowCollectorAddrToNetAddr?

Reply:
Could you just take input addr:port and proto as strings and do this when [creating FlowAggregator object](https://github.com/vmware-tanzu/antrea/blob/master/pkg/flowaggregator/flowaggregator.go#L153)? In the near future, go-ipfix will take these strings as inputs rather than net.Addr as input. In that case, we can remove these lines too.

Reply:
Got it, sounds good.